include/test/mat_graph_verifier.h

```diff
@@ -12,43 +12,24 @@
 class MatGraphVerifier : public GraphVerifier {
   std::vector<std::set<node_id_t>> kruskal_ref;
   std::vector<std::set<node_id_t>> boruvka_cc;
-  std::vector<bool>& det_graph;
+  std::vector<std::vector<bool>> adj_graph;
```

Comment:
is this the fix?

Reply:
It is most of it. The idea is to avoid the edge_id map and just use an adjacency matrix.